app/src/main/java/com/banuba/sdk/example/quickstart_c_api/MainActivity.kt

```diff
@@ -0,0 +1,214 @@
+        if (fileList?.isEmpty() == true) {
+            val file = File(baseFolder, path)
+            val parent = file.parentFile
+            if (!parent.exists() && !file.parentFile.mkdirs()) {
+                throw IOException(
+                    "Failed to create $parent. Check if you have `write` permissions"
+                )
+            }
+            application.assets.open(path).use { inputStream ->
+                FileOutputStream(file).use { outputStream ->
+                    processStreams(inputStream, outputStream)
+                }
+            }
+        } else {
+            if (fileList != null) {
+                for (children in fileList) {
+                    if (assetsToCopy.contains(children)) {
+                        val fullPath = File(path, children).path
+                        application.assets.list(fullPath)?.let {
+                            copyAssets(
+                                baseFolder,
+                                fullPath,
+                                it.toList()
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    @Throws(IOException::class)
+    private fun processStreams(inputStream: InputStream, outputStream: OutputStream) {
+        BufferedInputStream(inputStream).use { `in` ->
+            BufferedOutputStream(outputStream).use { out ->
+                val buffer = ByteArray(10240)
+                var bytesRead: Int
+                while (`in`.read(buffer).also { bytesRead = it } >= 0) {
+                    out.write(buffer, 0, bytesRead)
+                }
+            }
+        }
+    }
+}
+
+class EPRenderer(sdk: BanubaSdk, resPath: String, clientToken: String) : GLSurfaceView.Renderer {
+    private val banubaSdk = sdk
+
+    // Size of photo.jpg
+    private val width: Int = 1000
+    private val height: Int = 1500
+    private var frameNum = 0
+    private var byteBuffer: ByteBuffer? = null
+
+    private val oep: Long = banubaSdk.createOffscreenEffectPlayer(resPath, clientToken, width, height)
+    private var imageToProcess: Bitmap? = null
+
+    fun processImage(bitmap: Bitmap) {
+        imageToProcess = bitmap
+    }
+
+    override fun onSurfaceCreated(gl: GL10?, config: EGLConfig?) {
+        banubaSdk.loadEffect(oep, "effects/Afro")
+        banubaSdk.surfaceChanged(oep, width, height);
+    }
+
+    override fun onSurfaceChanged(gl: GL10, w: Int, h: Int) {
+        gl.glViewport(0, 0, width, height)
+    }
+
+    override fun onDrawFrame(gl: GL10) {
+        val bitmap = imageToProcess
+        if (byteBuffer == null) {
+            val size = bitmap!!.rowBytes * bitmap.height
+            byteBuffer = ByteBuffer.allocateDirect(size)
+            bitmap.copyPixelsToBuffer(byteBuffer)
+        }
+
+        if (byteBuffer != null && bitmap != null) {
+            banubaSdk.processImageAsync(oep, byteBuffer!!, bitmap.width, bitmap.height)
```

Comment:
maybe it is worth to consider hiding `oep` inside `banubaSdk` . it looks like unnecessary argument here